tools/cli/factory_test.go

```diff
@@ -0,0 +1,141 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cli
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"strings"
+	"testing"
+)
+
+type args struct {
+	path string
+}
+type testCase struct {
+	name    string
+	args    args
+	wantErr bool
+}
+
+type httpClientWrapper struct {
+	client  *http.Client
+	testUrl string
+}
+
+func (c *httpClientWrapper) Get(_ string) (resp *http.Response, err error) {
+	// mock all calls to GET using the testUrl instead
+	return c.client.Get(c.testUrl)
+}
+
+func Test_fetchCACertFromUrl(t *testing.T) {
+	// Ultimately, any URL like:
+	// https://example.com/testdata/4096b-rsa-example-cert.pem
+	// Will attempt to load the local file: testdata/4096b-rsa-example-cert.pem (removing
+	// the host), but "wrapped" in a httptest server
+	tests := []testCase{
```

Comment:
Can we add a test case in which we attempt to pull the cert from a server which itself has an invalid HTTPS certificate?  The current code should error out as written; I want to make sure it continues to.